lib/WorkerHandler.js

```diff
@@ -261,8 +262,9 @@ WorkerHandler.prototype.busy = function () {
  *                                  after finishing all tasks currently in
  *                                  progress. If true, the worker will be
  *                                  terminated immediately.
+ * @param {function} [callback=null] If provided, will be called when process terminates.
  */
-WorkerHandler.prototype.terminate = function (force) {
+WorkerHandler.prototype.terminate = function (force, callback) {
```

Comment:
Would it make sense to simply return a Promise instead of passing a callback? Most of the API is Promise based anyway :)

Reply:
Yep, in theory, that sounds great. I tried it, and I couldn't make it work. The problem, as best as I could tell, has to do with the fact that WorkerHandler.terminate is called in the 'message' and 'error' handlers. So the scenario is this:

Reply:
I think it should be possible to move the `var resolver = Promise.defer();` structure from `terminateAndNotify` to `terminate` and return a promise there. And both callback and Promise should resolve only once or course. It's not a big deal though, I'm ok with leaving this a callback structure (we can always do a refactoring later on if we want).